ibis/backends/sql/rewrites.py

```diff
@@ -161,14 +161,13 @@ def merge_select_select(_, **kwargs):
     from the inner Select are inlined into the outer Select.
     """
     # don't merge if either the outer or the inner select has window functions
-    for v in _.selections.values():
-        if v.find(Window, filter=ops.Value):
-            return _
-    for v in _.parent.selections.values():
-        if v.find((Window, ops.Unnest), filter=ops.Value):
-            return _
-    for v in _.predicates:
-        if v.find((ops.ExistsSubquery, ops.InSubquery), filter=ops.Value):
+
+    # TODO(kszucs): need a better api to find_under() a certain node without
+    # considering to filter the starting node out
+    for child in _.__children__ + _.parent.__children__:
+        if child.find(
+            (Window, ops.ExistsSubquery, ops.InSubquery, ops.Unnest), filter=ops.Value
```

Comment:
Really we should include scalar subqueries here as well since those can be correlated as well. The subqueries could be removed from here altogether if we don't stop traversing the predicates after reaching a non-value.